extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -109,6 +114,15 @@ public String getType()
     return TYPE;
   }
 
+  @Nonnull
+  @JsonIgnore
+  @Override
+  public Set<ResourceAction> getInputSourceResources()
+  {
+    // the input sources are properly computed in the SQL / calcite layer, but not in the native MSQ task here.
```

Comment:
Thanks for clarifying! One more reason why we cannot remove the default implementation is it's a method on a common interface and can break compilation because things can be loaded from custom extensions. So I think having the default behavior is good 👍